random1on1/api/algorithm.py

```diff
@@ -0,0 +1,18 @@
+import abc
+import networkx
+import numpy
+
+
+class MatchingAlgorithm(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        participant_graph: networkx.Graph[random1on1.api.participant.Participant],
```

Comment:
Not sure about only imputing the object of one Participant into the matching object.

On other note, I feel like we should create an instalable python package out of this.

```
from random1on1.api.participant import Participant
...
neparticipant_graph: networkx.Graph[Participant]
```